### PR TITLE
made improvements to several methods in taquito-utils

### PR DESCRIPTION
**Pull request by dsawali**
closes:
- #2372 
- #2274 

changes: 
- updated `bytes2char()` into `hexStringToByteString()`
- updated `char2bytes()` into `byteStringToHexString()`
- marked `char2bytes()` and `bytes2char()` as deprecated
- change `encodePubKey` name into `encodeAddress`
- updated `encodeAddress` (previously encodePubKey) to strip `0x` prefix in parameters

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [x] Your code builds cleanly without any errors or warnings
- [x] You have run the linter against the changes
- [x] You have added unit tests (if relevant/appropriate)
- [x] You have added integration tests (if relevant/appropriate)
- [x] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

In this PR, please also make sure: 

- [x] You have linked this PR to the issue by putting `closes #TICKETNUMBER` in the description box (when applicable)
- [x] You have added a concise description on your changes
## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: packages/taquito-utils/src/taquito-utils.ts [hunk 1]
```diff
@@ -133,6 +134,25 @@ export function encodePubKey(value: string) {
...
   return b58cencode(value.substring(2, 42), prefix.KT);
 }
+
+/**
+ *
+ * @description Base58 encode an address using predefined prefix (tz1, tz2, tz3). KT1 is not supported.
+ *
+ * @param value Address to base58 encode (tz1, tz2, tz3 or KT1)
+ */
+export function encodeAddress(value: string) {
```
Comment (hui-an-yang) [inline packages/taquito-utils/src/taquito-utils.ts:144]: My thoughts on this function [here](https://github.com/ecadlabs/taquito/issues/2372#issuecomment-1877902966), do you mind adding the strip '0x' logic and disclaim it's limitation with originated address with annots?

**Review by hui-an-yang — APPROVED** [review]
lgtm, just a minor enhancement comment to encodeAddress, will respect your choice if you prefer to do them in separate pr

File: docs/version.md [hunk 2]
```diff
@@ -314,13 +314,13 @@ const Tezos = new TezosToolkit(rpcUrl);
 
 Tezos.setWalletProvider(wallet)
 const signedW = await Tezos.wallet.signFailingNoop({
-arbitrary: char2Bytes("Hello World"),
+arbitrary: stringToBytes("Hello World"),
```
Comment (hui-an-yang) [inline docs/version.md:317]: might want to keep version.md as is, back in v17.2.0 release note char2Bytes was used

File: packages/taquito-utils/src/taquito-utils.ts (hunk 1 above)
Comment (dsawali) [reply packages/taquito-utils/src/taquito-utils.ts:144]: that's a good idea, I'll update

Comment (hui-an-yang) [reply packages/taquito-utils/src/taquito-utils.ts:144]: thanks much for the update, prior your change the description marked kt1 is not supported which isn't true, am hoping we can removed change the description to  

 * @description Base58 encode an address using predefined prefix tz1, tz2, tz3 or KT1(without annotation). 
 * @param value Address to base58 encode tz1, tz2, tz3 or KT1, supports value with and without prefix `0x`

Comment (dsawali) [reply packages/taquito-utils/src/taquito-utils.ts:144]: updated

**Pull request merged**